### ISSUE TITLE
Panic within acme.init on startup

### ISSUE THREAD

**Issue by kelunik** [containous/traefik#2953]
### What did you do?

We tried to start our setup using `docker-compose up`, but Traefik didn't startup as expected. The same setup worked two weeks ago. In the meantime, a new kernel has been installed, but that shouldn't result in the below error, should it?

We were pinned to `v1.5.0rc2` previously, tried a new container and also the newer `v1.5.2`, but same error there. The same `Dockerfile` works fine on other systems.

### What did you expect to see?

A successful startup.

### What did you see instead?

```
example-traefik      | panic: runtime error: slice bounds out of range
example-traefik      |
example-traefik      | goroutine 1 [running]:
example-traefik      | github.com/containous/traefik/vendor/github.com/miekg/dns.ClientConfigFromFile(0x211b1eb, 0x10, 0x0, 0x0, 0x0)
example-traefik      |      /go/src/github.com/containous/traefik/vendor/github.com/miekg/dns/clientconfig.go:86 +0xad6
example-traefik      | github.com/containous/traefik/vendor/github.com/xenolf/lego/acme.getNameservers(0x211b1eb, 0x10, 0x32e9920, 0x2, 0x2, 0x92d9bc, 0xc4202240e0, 0x1)
example-traefik      |      /go/src/github.com/containous/traefik/vendor/github.com/xenolf/lego/acme/dns_challenge.go:40 +0x4d
example-traefik      | github.com/containous/traefik/vendor/github.com/xenolf/lego/acme.init()
example-traefik      |      /go/src/github.com/containous/traefik/vendor/github.com/xenolf/lego/acme/dns_challenge.go:33 +0x12d
example-traefik      | github.com/containous/traefik/acme.init()
example-traefik      |      <autogenerated>:1 +0x84
example-traefik      | main.init()
example-traefik      |      <autogenerated>:1 +0x7f
example-web          | [Tue Feb 27 09:49:28.315122 2018] [mpm_prefork:notice] [pid 1] AH00163: Apache/2.4.10 (Debian) configured -- resuming normal operations
example-web          | [Tue Feb 27 09:49:28.315153 2018] [core:notice] [pid 1] AH00094: Command line: 'apache2 -D FOREGROUND'
example              | [Tue Feb 27 09:49:29.826700 2018] [mpm_event:notice] [pid 13:tid 140190014432448] AH00489: Apache/2.4.25 (Debian) configured -- resuming normal operations
example              | [Tue Feb 27 09:49:29.826804 2018] [core:notice] [pid 13:tid 140190014432448] AH00094: Command line: '/usr/sbin/apache2 -D FOREGROUND'
example-traefik exited with code 2
```

### Version

[Official Docker image](https://hub.docker.com/_/traefik/) v1.5.0rc2, but same with v1.5.2.

### Environment & Configuration

```toml
# Based on https://www.digitalocean.com/community/tutorials/how-to-use-traefik-as-a-reverse-proxy-for-docker-containers-on-ubuntu-16-04

defaultEntryPoints = ["http"]

[web]
address = ":8080"
```

**Comment by Juliens** [comment]
Thanks for your interest in Træfik.

Do you have some configurations about ACME in your toml file or in your command line arguments?

**Comment by kelunik** [comment]
@Juliens No, that's the strange thing. And it worked previously. Everything that's in the config is in the initial report. The relevant section of the `docker-compose.yml` is that:

```yml
  example-traefik:
    image: traefik:1.5.0-rc2
    container_name: example-traefik
    command: '--docker --entrypoints="Name:http Address::${EXAMPLE_PORT}"'
    ports:
      - '${EXAMPLE_PORT}:${EXAMPLE_PORT}'
    networks:
      example:
        aliases:
          - example-web.test
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - ./traefik/traefik.toml:/etc/traefik/traefik.toml
    environment:
      - EXAMPLE_PORT=${EXAMPLE_PORT}
    labels:
      traefik.backend: 'example-traefik'
      traefik.frontend.rule: 'Host:example-traefik.test'
      traefik.port: '8080'
```

**Comment by ldez** [comment]
@kelunik could you give the whole content of your `traefik.toml`.

**Comment by kelunik** [comment]
@ldez What I posted in the initial report really is the whole `traefik.toml`.

**Comment by ldez** [comment]
@kelunik sorry but it's not possible.
You must have a `[acme]` section.

**Comment by ldez** [comment]
@kelunik could you clean your `resolv.conf` ?

**Comment by traefiker** [comment]
Closed by #2955.
